lib/db_connection.ex

```diff
@@ -980,6 +1192,17 @@ defmodule DBConnection do
     end
   end
 
+  defp checkin(%DBConnection{} = conn, fun, meter, opts) do
+    with {:ok, conn_state, meter} <- fetch_info(conn, meter) do
+      return = fun.(conn, conn_state, meter, opts)
+      checkin(conn, opts)
+      return
+    end
+  end
+  defp checkin(pool, fun, meter, opts) do
+    run(pool, fun, meter, opts)
```

Comment:
I'm not sure what will happen in this case. Are we going to check out a connection, call `checkin` from the fun and then check the connection back in?

I'm not sure I understand the use case of calling `checkin` with the pool & without an actual connection.